src/lasp_dependence_dag.erl

```diff
@@ -44,20 +53,53 @@
 %% lets us quickly delete those edges.
 -type process_map() :: dict:dict(pid(), {id(), id()}).
 
+%% Stored metadata for a lasp process.
+%% We don't store the input(s) and output vertices as that information
+%% is implicitly stored by edges in the graph.
+-record(process_metadata, {read :: function(),
```

Comment:
Yes, this metadata is only saved for process that are removed during a path contraction,
meaning they link unnecessary vertices only. That's why they only have one read function.

Processes that have _n_ inputs are represented by _n_ edges, each one from one of the inputs to the output, with the appropiate read function as part of the edge label.